### PR TITLE
Make sure to log unhandled errors always

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Make sure to log unhandled errors always

## Motivation and Context
In many situations, while testing we encounter
ErrInternalError, to reduce logging we have
removed logging from quite a few places which
is acceptable but when ErrInternalError occurs
we should have a facility to log the corresponding
error, this helps to debug Minio server.

## Regression
No

## How Has This Been Tested?
There is no easier way to test this PR unless to generate an unhandled error. This is purely an enhancement to our logging needs. 
 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#5c375f92-52cd-450d-b061-7b10de9bea2d) |
<a name="5c375f92-52cd-450d-b061-7b10de9bea2d"></a>
### 6784-0e2447f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30683
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 38def438adf2:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 28 seconds
{
  "name": "awscli",
  "duration": 14107,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))"
}

Executed 0 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#2f4a822f-d17d-4d24-b637-09d598ddcd4f) |
<a name="2f4a822f-d17d-4d24-b637-09d598ddcd4f"></a>
### 6784-1cc602a/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31155
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 532efc6b9e3a:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 34 seconds
(2/13) Running aws-sdk-go tests ... done in 1 seconds
(3/13) Running aws-sdk-java tests ... done in 5 seconds
(4/13) Running aws-sdk-php tests ... done in 50 seconds
(5/13) Running aws-sdk-ruby tests ... done in 18 seconds
(6/13) Running mc tests ... done in 1 minutes and 26 seconds
(7/13) Running minio-dotnet tests ... FAILED in 6 minutes and 1 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Minio API responded with message=We encountered an internal error, please try again.) (Minio API responded with message=We encountered an internal error, please try again.)) ---> System.AggregateException: One or more errors occurred. (Minio API responded with message=We encountered an internal error, please try again.) (Minio API responded with message=We encountered an internal error, please try again.) ---> Minio.Exceptions.MinioException: Minio API responded with message=We encountered an internal error, please try again.
   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 363
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 505
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 351
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 483
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 229
   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 740
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.RemoveObjects_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 1738
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 218

Executed 6 out of 13 tests successfully.
```

File: cmd/admin-handlers.go [hunk 1]
```diff
@@ -107,8 +110,8 @@ func (a adminAPIHandlers) ServiceStatusHandler(w http.ResponseWriter, r *http.Re
 	// of read-quorum availability.
 	uptime, err := getPeerUptimes(globalAdminPeers)
 	if err != nil {
-		writeErrorResponseJSON(w, toAdminAPIErrCode(err), r.URL)
-		logger.LogIf(context.Background(), err)
+		writeErrorResponseJSON(w, toAdminAPIErrCode(ctx, err), r.URL)
+		logger.LogIf(ctx, err)
```
Comment (ebozduman) [inline cmd/admin-handlers.go]: `writeErrorResponseJSON` already sends the error to the client.
`logger.LogIf(ctx, err)` is not needed here.

File: cmd/admin-handlers.go [hunk 2]
```diff
@@ -145,7 +149,7 @@ func (a adminAPIHandlers) ServiceStopNRestartHandler(w http.ResponseWriter, r *h
 	var sa madmin.ServiceAction
 	err := json.NewDecoder(r.Body).Decode(&sa)
 	if err != nil {
-		logger.LogIf(context.Background(), err)
+		logger.LogIf(ctx, err)
```
Comment (ebozduman) [inline cmd/admin-handlers.go:151]: `writeErrorResponseJSON` already sends the error to the client.
`logger.LogIf(ctx, err)` is not needed here.

File: cmd/admin-handlers.go [hunk 3]
```diff
@@ -158,7 +162,7 @@ func (a adminAPIHandlers) ServiceStopNRestartHandler(w http.ResponseWriter, r *h
 		serviceSig = serviceStop
 	default:
 		writeErrorResponseJSON(w, ErrMalformedPOSTRequest, r.URL)
-		logger.LogIf(context.Background(), errors.New("Invalid service action received"))
+		logger.LogIf(ctx, errors.New("Invalid service action received"))
```
Comment (ebozduman) [inline cmd/admin-handlers.go:164]: `writeErrorResponseJSON` already sends the error to the client.
`logger.LogIf(ctx, errors.New("Invalid service action received"))` is not needed here.

File: cmd/admin-handlers.go [hunk 4]
```diff
@@ -926,7 +926,7 @@ func (a adminAPIHandlers) AddUser(w http.ResponseWriter, r *http.Request) {
 
 	if err = globalIAMSys.SetUser(accessKey, uinfo); err != nil {
 		logger.LogIf(ctx, err)
```
Comment (ebozduman) [inline cmd/admin-handlers.go]: `writeErrorResponseJSON` already sends the error to the client.
`logger.LogIf(ctx, err)` is not needed here.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#5da1fe3d-4749-4bdf-8b15-9efe183dea6d) |
<a name="5da1fe3d-4749-4bdf-8b15-9efe183dea6d"></a>
### 6784-15a8e15/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31542
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp beb9c8453e13:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 43 seconds
(2/13) Running aws-sdk-go tests ... done in 5 seconds
(3/13) Running aws-sdk-java tests ... done in 6 seconds
(4/13) Running aws-sdk-php tests ... done in 1 minutes and 17 seconds
(5/13) Running aws-sdk-ruby tests ... done in 46 seconds
(6/13) Running mc tests ... done in 2 minutes and 25 seconds
(7/13) Running minio-dotnet tests ... FAILED in 6 minutes and 37 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.)) ---> System.AggregateException: One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) ---> Minio.Exceptions.InternalClientException: Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.
   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 450
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 505
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 351
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 483
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 229
   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 740
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.RemoveObjects_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 1738
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 218

Executed 6 out of 13 tests successfully.
```

**Review by ebozduman — DISMISSED** [review]

File: cmd/admin-handlers.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/admin-handlers.go:164]: This is specific to his case so extra logging is needed...

File: cmd/admin-handlers.go (hunk 4 above)
Comment (harshavardhana) [reply cmd/admin-handlers.go]: This is outdated already @ebozduman its already fixed in lastest.

File: cmd/admin-handlers.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/admin-handlers.go:151]: This is needed as well we don't reply a correct error to client in this situation

File: cmd/admin-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/admin-handlers.go]: This is outdated, I have fixed all of this already @ebozduman

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :x: [more...](#f97d0d8b-6c88-4730-8a1b-e6ffb1550e8c) |
<a name="f97d0d8b-6c88-4730-8a1b-e6ffb1550e8c"></a>
### 6784-5e87ae1/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31291
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 24bfafc08ad5:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 21 seconds
{
  "name": "awscli",
  "duration": 12483,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```

**Review by ebozduman — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
ping @vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**